### PR TITLE
OpenStack: Add upgrade job testing 4.14.z -> master

### PR DESCRIPTION
**Pull request by dulek**
This commit adds an OpenStack job testing an upgrade from latest 4.14 nightly to the current 4.14 nightly.

### CODE REVIEW

File: ci-operator/config/shiftstack/shiftstack-ci/shiftstack-shiftstack-ci-main__periodic-4.14-upgrade-from-stable-4.14.yaml [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+  initial:
+    candidate:
+      product: ocp
+      relative: 1
+      stream: nightly
+      version: "4.14"
+  latest:
+    candidate:
+      product: ocp
+      stream: ci
```
Comment (mandre) [inline ci-operator/config/shiftstack/shiftstack-ci/shiftstack-shiftstack-ci-main__periodic-4.14-upgrade-from-stable-4.14.yaml:20]: shouldn't it be the nightly stream as well?

Looking at https://github.com/openshift/release/blob/master/ci-operator/config/openshift/release/openshift-release-master__nightly-4.14.yaml#L86-L97, they do nightly -> nightly upgrade.

**Comment by dulek** [comment]
/pj-rehearse

Looks like just etcd warnings.

File: ci-operator/config/shiftstack/shiftstack-ci/shiftstack-shiftstack-ci-main__periodic-4.14-upgrade-from-stable-4.14.yaml
Comment (jsafrane) [inline ci-operator/config/shiftstack/shiftstack-ci/shiftstack-shiftstack-ci-main__periodic-4.14-upgrade-from-stable-4.14.yaml:1]: should it be named `upgrade-from-stable` if it's from a nightly? IMO just `-upgrade` would be better.

Comment (dulek) [reply ci-operator/config/shiftstack/shiftstack-ci/shiftstack-shiftstack-ci-main__periodic-4.14-upgrade-from-stable-4.14.yaml:1]: Done!

**Comment by mandre** [comment]
/pj-rehearse periodic-ci-shiftstack-shiftstack-ci-main-periodic-4.14-upgrade-4.14-e2e-openstack-ovn-upgrade

**Comment by pierreprinetti** [comment]
Nice work, thanks. If you feel adventurous, you can try to update our CI-gitops: https://github.com/shiftstack/shiftstack-ci/blob/main/bump-periodics.sh

**Comment by pierreprinetti** [comment]
/pj-rehearse periodic-ci-shiftstack-shiftstack-ci-main-periodic-4.14-upgrade-4.14-e2e-openstack-ovn-upgrade periodic-ci-shiftstack-shiftstack-ci-main-periodic-4.14-upgrade-from-stable-4.14-e2e-openstack-ovn-upgrade

**Comment by dulek** [comment]
/pj-rehearse

This seems to be close.

**Comment by dulek** [comment]
/pj-rehearse periodic-ci-shiftstack-shiftstack-ci-main-periodic-4.14-upgrade-4.14-e2e-openstack-ovn-upgrade

**Comment by dulek** [comment]
/pj-rehearse periodic-ci-shiftstack-shiftstack-ci-main-periodic-4.14-upgrade-4.14-e2e-openstack-ovn-upgrade

**Comment by dulek** [comment]
140 tests failed, that's weird.

/pj-rehearse periodic-ci-shiftstack-shiftstack-ci-main-periodic-4.14-upgrade-4.14-e2e-openstack-ovn-upgrade

**Comment by dulek** [comment]
/pj-rehearse periodic-ci-shiftstack-shiftstack-ci-main-periodic-4.14-upgrade-4.14-e2e-openstack-ovn-upgrade

**Comment by dulek** [comment]
/pj-rehearse periodic-ci-shiftstack-shiftstack-ci-main-periodic-4.14-upgrade-4.14-e2e-openstack-ovn-upgrade

**Comment by dulek** [comment]
/pj-rehearse periodic-ci-shiftstack-shiftstack-ci-main-periodic-4.14-upgrade-4.14-e2e-openstack-ovn-upgrade

One last try and if it fails I drop this.

**Pull request closed without merging**